torch/distributed/_composable/fsdp/_fsdp_init.py

```diff
@@ -28,3 +34,50 @@ def _init_default_fully_shard_mesh(device_type: str) -> DeviceMesh:
     default_pg = dist.distributed_c10d._get_default_group()
     mesh = init_device_mesh(device_type=device_type, mesh_shape=(default_pg.size(),))
     return mesh
+
+
+def _get_managed_modules(root_module: nn.Module) -> List[nn.Module]:
+    modules: List[nn.Module] = []
+    # Track visisted modules to avoid visiting shared modules multiple times
+    visited_modules: Set[nn.Module] = set()
+
+    def dfs(module: nn.Module) -> None:
+        """
+        Runs a DFS to collect managed modules, not recursing into modules with
+        a non-composable API or ``fully_shard`` already applied.
+        """
+        if not _is_composable_with_fsdp(module):
+            return
+        elif module is not root_module and _get_module_fsdp_state(module) is not None:
+            return  # nested `fully_shard` module
+        visited_modules.add(module)
+        for submodule in module.children():
+            if submodule is not None and submodule not in visited_modules:
```

Comment:
Just out of curiosity, when will `children()` return None?

Reply:
Good point. In normal PyTorch, I think it cannot return `None` since `_named_members()` filters `None`. I am not sure about internal overrides. We can remove this check until we hit the error.